### PR TITLE
Immunization doseQuantity.system & doseQuantity.code update

### PR DESCRIPTION
**Pull request by bcastillomartel-va**
https://vajira.max.gov/browse/API-23195

### CODE REVIEW

**Comment by jblefkove-va** [comment]
Let's update `DatamartImmunization.Quantity` to remove `system` and `code`

(To support deserialization of sandbox payloads with those two fields, add setter methods to discard the values)

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java [hunk 1]
```diff
@@ -32,22 +32,14 @@ SimpleQuantity doseQuantity(Optional<DatamartImmunization.Quantity> maybeDoseQua
...
-          .unit(doseQuantity.unit().orElse(null))
-          .build();
-    }
     return SimpleQuantity.builder()
         .value(doseQuantity.value().orElse(null))
         .unit(doseQuantity.unit().orElse(null))
-        .system(system.get())
-        .code(code.get())
+        .system("http://unitsofmeasure.org")
+        .code(doseQuantity.unit().orElse(null))
```
Comment (jhulbert-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java:42]: If the code is null, we shouldn't have a system. That's a fhir requirement.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java [hunk 2]
```diff
@@ -32,22 +32,17 @@ SimpleQuantity doseQuantity(Optional<DatamartImmunization.Quantity> maybeDoseQua
...
-    if (allBlank(doseQuantity.value(), doseQuantity.unit(), system, code)) {
+    if (allBlank(doseQuantity.value(), doseQuantity.unit())) {
       return null;
     }
-    if (system.isEmpty() || code.isEmpty()) {
-      return SimpleQuantity.builder()
-          .value(doseQuantity.value().orElse(null))
-          .unit(doseQuantity.unit().orElse(null))
-          .build();
+    if (isBlank(doseQuantity.unit())) {
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java:38]: this check is unnecessary

File: data-query/src/test/resources/gov/va/api/health/dataquery/service/controller/immunization/datamart-immunization-v0.json [hunk 3]
```diff
@@ -16,8 +16,6 @@
   },
   "wasNotGiven" : false,
   "doseQuantity" : {
-    "code" : "ng/mL",
```
Comment (jblefkove-va) [inline data-query/src/test/resources/gov/va/api/health/dataquery/service/controller/immunization/datamart-immunization-v0.json:19]: Can we leave these fields in place and let the setters discard them?

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**